### PR TITLE
Update RELEASE instructions

### LINKED ISSUE #232: Update `RELEASE.MD`

### PR DESCRIPTION
**Pull request by embano1**
Fixes #232 

## Proposed Changes

- Update RELEASE instructions
- Add issue and PR templates

### Pre-review Checklist

- [ ] **At least 80% unit test coverage**
- [ ] **E2E tests** for any new behavior
- [ ] **Docs** for any user-facing impact

**Release Note**

```release-note
:broom: Update release guidance and add issue and PR templates
```

### CODE REVIEW

**Comment by embano1** [comment]
TODO:

- [ ] Add to PR template

File: .github/pull-request-template.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+### Pre-review Checklist
+
+<!-- If these boxes are not checked, you will be asked to complete these requirements or explain why they do not apply to your PR. -->
+
+- [ ] **At least 80% unit test coverage**
+- [ ] **E2E tests** for any new behavior
+- [ ] **Docs PR** for any user-facing impact
+- [ ] **Spec PR** for any new API feature
+- [ ] **Conformance test** for any change to the spec
```
Comment (evankanderson) [inline .github/pull-request-template.md:26]: I think these may be eventing-core specific?

File: .github/pull-request-template.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+```
+
+
+**Docs**
+
+<!--
+:book: If this change has user-visible impact, link to an issue or PR in
+https://github.com/knative/docs.
+-->
+
```
Comment (evankanderson) [inline .github/pull-request-template.md:50]: I'm assuming docs for this are in the same repo, so we shouldn't need this?

File: .github/pull-request-template.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+-
+-
+
+### Pre-review Checklist
+
+<!-- If these boxes are not checked, you will be asked to complete these requirements or explain why they do not apply to your PR. -->
+
+- [ ] **At least 80% unit test coverage**
+- [ ] **E2E tests** for any new behavior
+- [ ] **Docs PR** for any user-facing impact
```
Comment (evankanderson) [inline .github/pull-request-template.md:24]: Probably doesn't need a separate PR, maybe:

Docs updates for any user-facing changes

**Review by evankanderson — COMMENTED** [review]
You have a bunch of LICENSE files that you somehow marked `+x`, and a few comments on the templates (that look to be lifted from the eventing repo).

**Comment by embano1** [comment]
> You have a bunch of LICENSE files that you somehow marked `+x`

Where do you see those files in the PR?

**Comment by evankanderson** [comment]
> > You have a bunch of LICENSE files that you somehow marked `+x`
> 
> Where do you see those files in the PR?

"Verify Deps and Codegen" is failing.

**Comment by embano1** [comment]
> "Verify Deps and Codegen" is failing.

Yes, but this is not due to this PR (no vendored changes) so this should be good to merge and a separate PR might have to fix these.

**Review by n3wscott — APPROVED** [review]

**Pull request merged**